app/config/collections.php

```diff
@@ -479,6 +479,17 @@
                 'array' => false,
                 'filters' => [],
             ],
+            [
+                '$id' => ID::custom('provider'),
+                'type' => Database::VAR_STRING,
+                'format' => '',
+                'size' => 128,
+                'signed' => true,
+                'required' => false,
+                'default' => null,
+                'array' => false,
+                'filters' => [],
+            ],
```

Comment:
@stnguyen90 We had to create a new field for the tokens collection named `provider`. I tried to do it without creating a new field but this seems inevitable.

So, the reason for this is -

Here - https://github.com/appwrite/appwrite/pull/8389/files#diff-93771e9bb8e57cbe7d0c0e91e7aea5d85238a362f51caf957049f1d2ccc4d312R81, where we are fetching the identity of this particular user for this particular provider, we currently didn't have the oauth2 provider name.
So, we have multiple identities for a user. So, how would we get the identity for that particular oauth2 provider?
Because of that only we needed to have that information and as per I think it will make sense if this data comes from the token itself.

But I am unsure how the existing projects will adapt this, because else it will give `DOCUMENT_INVALID_STRUCTURE` at the early stage when creating the oauth2 token only.
Is there some mechanism implemented to adapt these collection changes to already created projects/databases?

Reply:
@Souptik2001, how about we pull the provider* values from the identity?

https://github.com/appwrite/appwrite/blob/09664c7e60f22ecaa980d3fe4cac612544db0923/app/controllers/api/account.php#L1430-L1451

Reply:
@Souptik2001, how about we pull the provider* values from the identity?

https://github.com/appwrite/appwrite/blob/09664c7e60f22ecaa980d3fe4cac612544db0923/app/controllers/api/account.php#L1430-L1451